sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ShiftDynamicNotInteger.CSharp9.cs

```diff
@@ -0,0 +1,15 @@
+﻿dynamic d = 5;
+var x = d >> 5.4; // Noncompliant
+
+nuint i = 0;
+var y = i >> 4; // Compliant
+
+nint j = 1;
+var z = j >> 4; // Compliant
+
+x = d >> new MyRecord();
```

Comment:
`d` is dynamic but this is not a false negative since `MyRecord` has an implicit cast to 1.
I've added one more test case since the current example is not very clear.